source/iface/accessor-concept.rst

```diff
@@ -2,41 +2,400 @@
+* Those accessors which provide access to data from host code do
+  block until the data is available on the host.
+
+.. warning::
+
+  For those accessors which provide access to data within a command,
+  the member functions which access data should only be called from
+  within the command. Programs which call these member functions
+  from outside of the command are ill formed.
+
+.. seealso:: |SYCL_SPEC_ACCESSORS|
+
+==========
+Data types
+==========
+
+All accessors have a ``DataT`` template parameter which specifies
+the type of each element that the accessor accesses. For
+:ref:`command-accessor` and :ref:`host_accessor`, this type must
+either match the type of each element in the underlying :ref:`buffer`,
+or it must be a ``const`` qualified version of that type.
+
+For the image accessors (:ref:`unsampled-image-accessors`,
+:ref:`sampled-image-accessors`), ``DataT`` must be one of:
+
+* ``sycl::int4`` (``sycl::vec<int32_t, 4>``);
+* ``sycl::uint4`` (``sycl::vec<uint32_t, 4>``);
+* ``sycl::float4`` (``sycl::vec<float, 4>``);
+* ``sycl::half4`` (``sycl::vec<half, 4>``).
+
+See :ref:`local_accessor` page for it's allowed ``DataT`` types.
+
 .. _access-mode:
 
-======================
-``sycl::access::mode``
-======================
+=====================
+``sycl::access_mode``
+=====================
+
+Most accessors have an ``AccessMode`` template parameter which
+specifies whether the accessor can read or write the underlying data.
+
+This information is used by the runtime when defining the requirements
+for the associated command, and it tells the runtime whether data
+needs to be transferred to or from a device before data can be
+accessed through the accessor.
+
+.. note::
+
+  Not all accessor classes support all modes, so see the description of each class for more details.
 
 ::
 
-  enum class mode {
-    read = 1024,
+  namespace sycl {
+
+  enum class access_mode : /* unspecified */ {
+    read,
     write,
     read_write,
-    discard_write,
-    discard_read_write,
-    atomic
+    discard_write,      // Deprecated in SYCL 2020
+    discard_read_write, // Deprecated in SYCL 2020
+    atomic              // Deprecated in SYCL 2020
   };
 
-.. _access-target:
+  namespace access {
+  // The legacy type "access::mode" is deprecated.
+  using mode = sycl::access_mode;
+  } // namespace access
+
+  } // namespace sycl
 
-========================
-``sycl::access::target``
-========================
+.. list-table::
+  :header-rows: 1
+
+  * - Access mode
+    - Description
+  * - ``sycl::access_mode::read``
+    - Read-only access.
+  * - ``sycl::access_mode::write``
+    - Write-only access.
+  * - ``sycl::access_mode::read_write``
+    - Read and write access.
+
+.. _deduction-tags:
+
+==============
+Deduction tags
+==============
+
+Some accessor constructors take a ``TagT`` parameter, which is
+used to deduce template arguments for the constructor's class.
+
+Each of the access modes in table above has an associated tag,
```

Comment:
```suggestion
Each of the access modes in the table above has an associated tag,
```

Reply:
Fixed in the https://github.com/KhronosGroup/SYCL_Reference/pull/96/commits/7544af73fd933d50fd209c31178a4e59475a7c93.

Reply:
Fixed in the https://github.com/KhronosGroup/SYCL_Reference/pull/96/commits/7544af73fd933d50fd209c31178a4e59475a7c93.